code/obj/item/instruments.dm

```diff
@@ -105,10 +101,59 @@
 	proc/post_play_effect(mob/user as mob)
 		return
 
+
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui && use_new_interface)
+			ui = new(user, src, "MusicInstrument")
+			ui.open()
+
+	ui_data(mob/user)
+		. = ..()
+		if (.)
+			return
+		. = list(
+			"name" = src.name,
+			"notes" = src.notes,
+		)
+
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		. = ..()
+		if(.)
+			return
+		switch(action)
+			if("play_note")
+				var/note_to_play = params["note"]
+				var/volume = params["volume"]
+				playsound(get_turf(src), sounds_instrument[note_to_play], volume, randomized_pitch, pitch = pitch_set)
+				. = FALSE
+			if("play_keyboard_on")
+				usr.client.apply_keybind("instrument_keyboard")
+				. = FALSE
+			if("play_keyboard_off")
+				usr.client.mob.reset_keymap()
+				. = FALSE
+
+	ui_close(mob/user)
+		user.reset_keymap()
+		. = ..()
+
+	ui_status(mob/user, datum/ui_state/state)
+		if(!IN_RANGE(src, user, 1)){
+			user.reset_keymap()
+			return UI_CLOSE
+		}
```

Comment:
No need for the `{}` here.
Also hm. Maybe also reset the keymap if `..()` returns UI_CLOSE for some reason? Not sure if that can happen here but better safe than sorry.